lib/kitchen/driver/azurerm.rb

```diff
@@ -334,15 +337,24 @@ def create(state)
         end
       end
 
+      # Return a True of False if the state is already stored for a particular property.
+      #
+      # @param [Hash] Hash of existing state values.
+      # @param [String] A property to check
+      # @return [Boolean]
       def existing_state_value?(state, property)
         state.key?(property) && !state[property].nil?
       end
 
+      # Leverage existing state values or bring state into existence from a configuration file.
+      #
+      # @param [Hash] Existing Hash of state values.
+      # @return [Hash] Updated Hash of state values.
       def validate_state(state = {})
```

Comment:
Yeah, I would recommend renaming the function altogether, since it really doesn't do any sort of validation at this point; it just populates the state with values from the `config` if they are not already set.

Reply:
Suggestions? 

Also, I applied your patch and after an initial test login failed for the machine. I'm comparing to the previous version to see what might have happened. No worries. We'll 'git er dun'.

Reply:
Perhaps `initialize_state`? Though the function is only used once, in this same file, so the only real value it currently provides is hiding implementation details (abstraction), and given that it has a misleading name I would argue that it's almost providing negative value from that perspective. If it were my code, I would probably just inline this function's contents and remove it.

Regarding `kitchen login`, can you show me the `.kitchen.yml` you used? I was unable to repro using [the `.kitchen.yml` from the relevant issue](https://github.com/test-kitchen/kitchen-azurerm/issues/118#issue-517447060).
I'm also in `chefcommunity.slack.com` if you'd prefer to chat in Slack.

I created a new empty directory `test`, created a `.kitchen.yml` containing the contents from my issue repro above and a `Gemfile` containing the following:
```
gem 'test-kitchen'
gem 'kitchen-azurerm', path: '../../github/kitchen-azurerm' # replace with the path to your local copy of the repository
gem 'kitchen-inspec'
gem 'berkshelf'
```

I then executed the following commands:
```
> bundle install
> bundle exec kitchen create # fails as expected, since ENV['KITCHEN_VM_PASSWORD'] is not set
> $env:KITCHEN_VM_PASSWORD = 'Password1234!' # I'm running in Windows 10 1903, PowerShell 5.1
> bundle exec kitchen create # succeeds as expected, since #118 is now fixed
> bundle exec kitchen login # opens Remote Desktop; I was able to login using the above password
```

I can provide the resulting `Gemfile.lock` to verify that we're both using the same versions of dependencies. My `chef --version` is:
```
Chef Workstation version: 20.6.62
Chef Infra Client version: 16.1.16
Chef InSpec version: 4.19.0
Chef CLI version: 3.0.4
Test Kitchen version: 2.5.1
Cookstyle version: 6.7.3
```